Library/Homebrew/language/haskell.rb

```diff
@@ -92,5 +92,27 @@ def install_cabal_package(*args, **options)
         end
       end
     end
+    module Stack
+      # run stack install, using the current latest LTS snapshot on stackage.org
+      def stack_install(*args)
+        system "stack", "install", "--resolver=lts", *args
```

Comment:
`lts` is a moving target, so this will have almost the same problem as the current `cabal` approach.  Instead, the `stack.yaml` provided by the package should always be used.

Reply:
Also, running `stack` like this will make it read from and update various things in `$HOME/.stack`.  I think it is necessary to set `STACK_HOME` to the build directory first.